src/pylibs/db/sql_adapter.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from sqlalchemy import create_engine, MetaData, Table, Column, String, Date, LargeBinary, Boolean
+import bcrypt
+import logging
+import random
+from datetime import datetime
+
+# timestamp format for birthdays
+TIMESTAMP_FORMAT = '%Y-%m-%d'
+
+
+class SqlAdapter:
+    def __init__(self, uri):
+        # connect to db
+        self.engine = create_engine(uri)
+        self.users_table = Table(
+            'users',
+            MetaData(self.engine),
+            Column('accountid', String),
+            Column('username', String),
+            Column('passhash', LargeBinary),
+            Column('firstname', String),
+            Column('lastname', String),
+            Column('birthday', Date),
+            Column('timezone', String),
+            Column('address', String),
+            Column('state', String),
+            Column('zip', String),
+            Column('ssn', String),
+        )
+        self.contacts_table = Table(
+            'contacts',
+            MetaData(self.engine),
+            Column('username', String),
+            Column('label', String),
+            Column('account_num', String),
+            Column('routing_num', String),
+            Column('is_external', Boolean),
+        )
+        # open a connection
+        self.conn = self.engine.connect()
+
+    def add_user(self, user):
```

Comment:
(I know you just copied the existing code, so apologies for the following.)

To be clear, as the code is written, this `user` parameter is actually a collection of attributes from an HTML form, including fields like `password` and `password-repeat`.

It also means that `accountid` needs to be added and `password` needs to be hashed into `passhash` before this object starts looking like an actual `user` object.  This sql adaptor class is meant to be a thin database wrapper and should probably not need to responsible for all that.

Suggestion:
(strong) move the password hashing back into `userservice.py`
(maybe) move accountid generation back into `userservice.py`
- would likely require implementing `sql_adapter.get_user(accountid)`
- This one is a little questionable and I could see the case for leaving it here in the adaptor because SQL is often left to generate object ids with `GENERATED AS IDENTITY`.  (We do it manually so that our user accounts don't look like `1000000001`, `1000000002`, `1000000003`, etc.)

Reply:
I have moved password hashing back into `userservice.py`
I havent moved account id generation back to `userservice.py` yet although I okay with implementing `sql_adapter.get_user(accountid)` and doing this. 
@Daniel-Sanche any thoughts?